js/oceanex.js

```diff
@@ -0,0 +1,617 @@
+            'quoteVolume': undefined,
+            'info': ticker,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let request = {
+            'market': market['id'].toLowerCase (),
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.publicGetOrderBook (this.extend (request, params));
+        const orderbook = this.safeValue (response, 'data');
+        let timestamp = this.safeInteger (orderbook, 'timestamp');
+        if (timestamp !== undefined) {
+            timestamp = timestamp * 1000;
+            // Get milliseconds
+        }
+        return this.parseOrderBook (orderbook, timestamp);
+    }
+
+    async fetchOrderBooks (symbols, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let markets = [];
+        for (let i = 0; i < symbols.length; i++) {
+            let marketId = this.marketId (symbols[i]);
+            markets.push (marketId.toLowerCase ());
+        }
+        const response = await this.publicGetOrderBookMulti (this.extend ({
+            'markets': markets,
+            'limit': limit,
+        }));
+        const data = this.safeValue (response, 'data');
+        let result = [];
+        for (let i = 0; i < data.length; i++) {
+            let orderbook = data[i];
+            let marketId = this.safeValue (orderbook, 'market');
+            marketId = marketId.toUpperCase ();
+            let market = this.markets_by_id[marketId];
+            let symbol = market['symbol'];
+            let timestamp = this.safeInteger (orderbook, 'timestamp');
+            if (timestamp !== undefined) {
+                timestamp = timestamp * 1000;
+                // Get milliseconds
+            }
+            result[symbol] = this.parseOrderBook (orderbook, timestamp);
+        }
+        return result;
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let request = {
+            'market': market['id'].toLowerCase (),
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.publicGetTrades (this.extend (request, params));
+        const data = this.safeValue (response, 'data');
+        return this.parseTrades (data, market, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        let side = this.safeValue (trade, 'side');
+        if (side === 'bid') {
+            side = 'buy';
+        } else if (side === 'ask') {
+            side = 'sell';
+        }
+        let symbol = undefined;
+        let marketId = this.safeValue (trade, 'market');
+        if (marketId !== undefined) {
+            marketId = marketId.toUpperCase ();
```

Comment:
This line ↑ `marketId.toUpperCase()` is not needed here as well if we edit `fetchMarkets` above.